extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -819,28 +812,67 @@ angular.module('oppia').component('svgFilenameEditor', {
         });
 
         ctrl.canvas.on('selection:created', function() {
+          ctrl.fillPicker.setOptions({
+            color: ctrl.canvas.getActiveObject().get('fill')
+          });
+          ctrl.strokePicker.setOptions({
+            color: ctrl.canvas.getActiveObject().get('stroke')
+          });
           ctrl.enableRemoveButton = true;
           if (ctrl.canvas.getActiveObject().get('type') === 'textbox') {
             ctrl.displayFontStyles = true;
           }
+          $scope.$apply();
+        });
+
+        ctrl.canvas.on('selection:updated', function() {
+          ctrl.fillPicker.setOptions({
+            color: ctrl.canvas.getActiveObject().get('fill')
+          });
+          ctrl.strokePicker.setOptions({
+            color: ctrl.canvas.getActiveObject().get('stroke')
+          });
+          if (ctrl.canvas.getActiveObject().get('type') === 'textbox') {
+            ctrl.displayFontStyles = true;
+          }
+          $scope.$apply();
         });
 
         ctrl.canvas.on('selection:cleared', function() {
           ctrl.enableRemoveButton = false;
           ctrl.displayFontStyles = false;
         });
+        $scope.$apply();
+      };
+
+      ctrl.setCanvasDimensions = function() {
+        ctrl.canvasContainer = document.getElementById(ctrl.canvasContainerId);
+        var width = ctrl.canvasContainer.offsetWidth;
+        var height = ctrl.canvasContainer.offsetHeight;
+        // Set the size of the canvas to be equal to that of the
+        // parent element only if it is greater than the current
+        // canvas size.
+        if (ctrl.canvasMaxHeight < height) {
+          ctrl.canvas.setHeight(height);
+          ctrl.canvasMaxHeight = height;
+        }
+        if (ctrl.canvasMaxWidth < width) {
+          ctrl.canvas.setWidth(width);
+          ctrl.canvasMaxWidth = width;
+        }
+        ctrl.canvas.renderAll();
+        $scope.$apply();
       };
 
       var initializeFabricJs = function() {
-        ctrl.canvasElement = document.getElementById(ctrl.canvasID);
-        // Make it visually fill the positioned parent
-        ctrl.canvasElement.style.width = '100%';
-        ctrl.canvasElement.style.height = '100%';
-        // ...then set the internal size to match
-        ctrl.canvasElement.width = ctrl.canvasElement.offsetWidth;
-        ctrl.canvasElement.height = ctrl.canvasElement.offsetHeight;
-
         ctrl.canvas = new fabric.Canvas(ctrl.canvasID);
+        ctrl.setCanvasDimensions();
+
+        ctrl.resizeSubscription = WindowDimensionsService.getResizeEvent().
+          subscribe(evt => {
+            ctrl.setCanvasDimensions();
```

Comment:
You might need to add a $rootScope.$applyAsync() method here as well. Can you check other usages of this?

Reply:
Done.

Reply:
Oh, sorry, I meant to add it in the subscribe block here and not in `setCanvasDimensions`. Since it is required as part of the resize event and not everytime that function is called.

Reply:
Done.